validation/pipeline/scripts/build_qa_infra.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+cd  /root/go/src/github.com/rancher/qa-infra-automation
+
+# Set variables
+REPO_ROOT=$(pwd)
+WORKSPACE_NAME="jenkins_workspace"
+TERRAFORM_DIR="terraform/aws/cluster_nodes"
+RKE2_PLAYBOOK_PATH="ansible/rke2/rke2-playbook.yml"
+TERRAFORM_INVENTORY="ansible/rke2/terraform-inventory.yml"
+ANSIBLE_CONFIG="ansible/rke2/ansible.cfg"
+RANCHER_PLAYBOOK_PATH="ansible/rancher/rancher-playbook.yml"
+TFVARS_FILE="cluster.tfvars"
+KUBECONFIG_FILE="$REPO_ROOT/ansible/rke2/kubeconfig.yaml"
+VARS_FILE="./ansible/vars.yaml"
+PRIVATE_KEY_FILE="/root/.ssh/jenkins-elliptic-validation.pem"
+
+# --- Terraform Steps ---
+terraform -chdir="$TERRAFORM_DIR" init -input=false
+
+# Create and select the Terraform workspace
+terraform -chdir="$TERRAFORM_DIR" workspace new "$WORKSPACE_NAME" || terraform -chdir="$TERRAFORM_DIR" workspace select "$WORKSPACE_NAME"
+
+# Export the TF_WORKSPACE environment variable
+export TF_WORKSPACE="$WORKSPACE_NAME"
+
+# Export the ANSIBLE_CONFIG environment variable
+export ANSIBLE_CONFIG="$ANSIBLE_CONFIG"
+
+# Export the ANSIBLE_CONFIG environment variable
+export ANSIBLE_PRIVATE_KEY_FILE="$PRIVATE_KEY_FILE"
+
+# Apply the Terraform configuration
+terraform -chdir="$TERRAFORM_DIR" apply -auto-approve -var-file="$TFVARS_FILE"
+if [ $? -ne 0 ]; then
+    echo "Error: Terraform apply failed."
+    exit 1
+fi
+
+# --- RKE2 Playbook ---
```

Comment:
I'm almost wondering if the playbooks should be its own separate script (and stage in the Jenkinsfile) so that it's easier to create other jobs in the future in case we want anything that just builds infra or just deploys on provided infra. Right now I don't think it's a big deal since it's currently built kind of coupled, but as we decouple to add more providers to the qa-infra-automation, it will make more sense to have these decouple as well.